internal/config/config.go

```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/influxdata/telegraf/plugins/inputs"
 	"github.com/influxdata/telegraf/plugins/outputs"
 
-	"github.com/naoina/toml"
+	"github.com/influxdata/config"
 	"github.com/naoina/toml/ast"
```

Comment:
I gave this a try this morning, and this isn't going to be as easy as I thought. For this type assertion to pass: https://github.com/influxdata/telegraf/blob/tr-config/internal/config/config.go#L451, I need to make our fork of `naoina/toml` produce `*config.Table`s instead of its own `*ast.Tables` because Go doesn't consider those types to be identical. This gets us into very strange territory. As much as I'd like to :scissors: this dependency, I don't know if the juice is worth the squeeze.